### ISSUE TITLE
We continue to have problems deploying and updating our sites.

### ISSUE THREAD

**Issue by eliseodedios** [sintaxi/surge#533]
We continue to have problems deploying and updating our sites.

**Comment by sseeaann** [comment]
Yes there seems to be some caching involved as updates are not reflected and requires many push attempts or even some sort of HTML element or CSS dom change to force it.

**Comment by sintaxi** [comment]
I'll look into it. First I need to know the IP address you are directed to. Can you run `ping <your-domain>` for me?

**Comment by eliseodedios** [comment]
ping rc-commander.surge.sh
PING rc-commander.surge.sh (138.197.235.123): 56 data bytes
64 bytes from 138.197.235.123: icmp_seq=0 ttl=47 time=64.671 ms
64 bytes from 138.197.235.123: icmp_seq=1 ttl=47 time=70.202 ms
64 bytes from 138.197.235.123: icmp_seq=2 ttl=47 time=64.308 ms
64 bytes from 138.197.235.123: icmp_seq=3 ttl=47 time=70.628 ms
64 bytes from 138.197.235.123: icmp_seq=4 ttl=47 time=72.393 ms
64 bytes from 138.197.235.123: icmp_seq=5 ttl=47 time=64.729 ms
64 bytes from 138.197.235.123: icmp_seq=6 ttl=47 time=68.290 ms

**Comment by sintaxi** [comment]
@eliseodedios thank you. 

We have purged caches on the 138.197.235.123 edgenode. please test again.

**Comment by eliseodedios** [comment]
@sintaxi Thanks for the help my dear, greetings and success.

**Comment by coolov** [comment]
hello! also experiencing staleness. my ip is 138.197.235.123

**Comment by darkralts** [comment]
hello i am having an issue teardown'ing one of my sites, and my other site wont update even though it says success! and surge command said it tore down site, when i go to my sites they are not updating and the other teard down one is still up

**Comment by jonathanconway** [comment]
Hi there, thanks so much for this awesome service!

However unfortunately I'm having the same issues described above.
Only the first uploaded version of my site is being served, despite making changes and uploading using the surge command.
Website: https://testmatic.surge.sh/

Would much appreciate any help. 🙏 
(Also please let me know if I can help you out somehow, I have some free time.)

Thanks!

**Comment by sintaxi** [comment]
@jonathanconway Im seeing 3 deployments all with identical content. are you sure you changed the contents of the project?

https://1719938981217-testmatic.surge.sh/auto.appcache
https://1719939578528-testmatic.surge.sh/auto.appcache
https://1719974067606-testmatic.surge.sh/auto.appcache

**Comment by jonathanconway** [comment]
@sintaxi Ok thanks for letting me know, I'll investigate on my side. Thanks!

**Comment by jonathanconway** [comment]
@sintaxi I rebuilt the site and checked my ./build folder. It includes the following files:

* static/js/main.a831ab0e.js
* static/css/main.c9ec1d83.css

These files also appear in the asset-manifest.json file:

```json
{
  "files": {
    "main.css": "/static/css/main.c9ec1d83.css",
    "main.js": "/static/js/main.a831ab0e.js",
    "static/js/453.997098d3.chunk.js": "/static/js/453.997098d3.chunk.js",
    "static/media/marked.cjs": "/static/media/marked.ce1e881bd4176f4ee7e9.cjs",
    "static/media/logo.svg": "/static/media/logo.2af48dab3fba65081f3f9d97088a78cb.svg",
    "index.html": "/index.html",
    "main.c9ec1d83.css.map": "/static/css/main.c9ec1d83.css.map",
    "main.a831ab0e.js.map": "/static/js/main.a831ab0e.js.map",
    "453.997098d3.chunk.js.map": "/static/js/453.997098d3.chunk.js.map"
  },
  "entrypoints": [
    "static/css/main.c9ec1d83.css",
    "static/js/main.a831ab0e.js"
  ]
}
```

I'm running this command for the deployment: 

```bash
surge ./build testmatic.surge.sh
```

But when I browse to the new file: https://testmatic.surge.sh/static/js/main.a831ab0e.js, I get a "Page not found" error.

When I browse to the older deployed file: https://testmatic.surge.sh/static/js/main.a831ab0e.js, I get served the file.

So it seems surge is caching the older version of the site, unless I'm missing something?

Version of surge I'm using: v0.23.1.

**Comment by sintaxi** [comment]
can you run `ls ./build` from the same directory?

**Comment by sseeaann** [comment]
I am plussing this issue. I have to push my updates 5-10 times before they take and are reflected on the site OR they don't update ever. For example, yesterday and the days before (since issue #530 was closed out), it was requiring me to push a few times before the site would be reflected. Today however, I've pushed a dozen times and the update is never reflected.

**Comment by mr3wi** [comment]
> I am plussing this issue. I have to push my updates 5-10 times before they take and are reflected on the site OR they don't update ever. For example, yesterday and the days before (since issue #530 was closed out), it was requiring me to push a few times before the site would be reflected. Today however, I've pushed a dozen times and the update is never reflected.

I'm experimenting the same issues on my side

**Comment by sintaxi** [comment]
@jonathanconway @sseeaann @mr3wi I need to know what endpoint you are being directed to. Can you run `ping <yoursubdomain>.surge.sh`?

**Comment by jonathanconway** [comment]
```
PING testmatic.surge.sh (139.59.195.30): 56 data bytes
64 bytes from 139.59.195.30: icmp_seq=0 ttl=51 time=111.415 ms
^C
--- testmatic.surge.sh ping statistics ---
6 packets transmitted, 6 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 104.192/106.844/111.415/2.498 ms
```

**Comment by jonathanconway** [comment]
It seems to be working again. Many thanks!! 👍
